var/spack/repos/builtin/packages/arborx/package.py

```diff
@@ -102,20 +105,22 @@ class Arborx(CMakePackage, CudaPackage, ROCmPackage):
     depends_on("trilinos@14.4.0:", when="@1.6:+trilinos")
     patch("trilinos14.0-kokkos-major-version.patch", when="@1.4+trilinos ^trilinos@14.0.0")
     conflicts("~serial", when="+trilinos")
-    conflicts("+cuda", when="+trilinos")
 
     def cmake_args(self):
         spec = self.spec
 
+        if "~trilinos" in spec:
+            kokkos_spec = spec["kokkos"]
+        else:
+            kokkos_spec = spec["trilinos"]
+
         options = [
-            "-DKokkos_ROOT=%s"
-            % (spec["kokkos"].prefix if "~trilinos" in spec else spec["trilinos"].prefix),
+            "-DKokkos_ROOT=%s" % (kokkos_spec.prefix),
             self.define_from_variant("ARBORX_ENABLE_MPI", "mpi"),
         ]
 
         if "+cuda" in spec:
-            # Only Kokkos allows '+cuda' for now
-            options.append("-DCMAKE_CXX_COMPILER=%s" % spec["kokkos"].kokkos_cxx)
+            options.append("-DCMAKE_CXX_COMPILER=%s" % kokkos_spec.kokkos_cxx)
         if "+rocm" in spec:
             options.append("-DCMAKE_CXX_COMPILER=%s" % spec["hip"].hipcc)
```

Comment:
@aprokop I believe this line is unnecessary now. I know Kokkos required the use of hipcc in the past, but that requirement was removed and for a while now we've been building with amdclang++ directly. I'd like that workflow to be possible (which just deleting these 2 lines enables), but I don't want to break anyone else's workflow. Thoughts on the best approach?

Reply:
That's a good point. I'm not sure. I wonder if we should remove all clauses and just always take the compiler from Kokkos/Trilinos
```
options.append("-DCMAKE_CXX_COMPILER=%s" % spec["kokkos"].kokkos_cxx)
```
or something like that. However, looking at the `kokkos-kernels` spec, it still have the same `hipcc` thingie. It was introduced outside of the development team, however, so I'm not 100% confident it is the right thing to do.
Should ask some people responsible for Kokkos packages.

I'm going to ask some Kokkos developers responsible for Spack maintenance.

Reply:
Did you get a response from the Kokkos developers on this? Anything else outstanding we need to get this merged?